cmd/fleet/apple_mdm.go

```diff
@@ -143,7 +143,7 @@ func registerMDM(
 	var mdmService nanomdm_service.CheckinAndCommandService = nanomdm.New(mdmStorage, nanomdm.WithLogger(mdmLogger))
 	mdmService = certauth.New(mdmService, mdmStorage)
 	var mdmHandler http.Handler = httpmdm.CheckinAndCommandHandler(mdmService, mdmLogger.With("handler", "checkin-command"))
-	mdmHandler = MDMHostIngesterMiddleware(mdmHandler, mdmHostIngester, logger)
+	mdmHandler = MDMHostIngesterMiddleware(mdmHandler, ds, logger)
```

Comment:
Maybe we should rename this to `MDMCheckinMiddleware` or something else?

Reply:
good idea, pushed a tweak in [2f84d1b](https://github.com/fleetdm/fleet/pull/9127/commits/2f84d1b7e630e87e5379ffd6fe0a64c807679210)